core/src/main/java/org/apache/iceberg/IncrementalFileCleanup.java

```diff
@@ -81,12 +75,11 @@ public void cleanFiles(TableMetadata beforeExpiration, TableMetadata afterExpira
       return;
     }
 
-    SnapshotRef branchToCleanup = Iterables.getOnlyElement(beforeExpiration.refs().values(), null);
-    if (branchToCleanup == null) {
+    Snapshot latest = beforeExpiration.currentSnapshot();
+    if (latest == null) {
```

Comment:
I think I had originally implemented this so that incremental cleanup can be performed on singular non-main branches (i.e. main is empty and there's some `test` branch which got cleaned up) but I do think that's a pretty narrow case to optimize for. 

I think going back to just always taking the latest on main on the metadata before expiration and deriving ancestry from that for incremental cleanup is safest and simplest.